receiver/apachesparkreceiver/factory_test.go

```diff
@@ -41,7 +48,12 @@ func TestNewFactory(t *testing.T) {
 						InitialDelay:       time.Second,
 					},
 					ClientConfig: confighttp.ClientConfig{
```

Comment:
Shouldn't this follow the same form as other changes, a call to `confighttp.NewDefaultClientConfig()`, then set values that we have defaults for, then use that created default config here?

Reply:
I've only been changing to `confighttp.NewDefaultClientConfig()` in non test files, IMO its clearer in tests to have the expected config as manually built struct.

Reply:
Doesn't that leave open the possibility of invalid tests (the functional code has a different config than tests), or tests that will just need updated every time default values change?

Reply:
> Doesn't that leave open the possibility of invalid tests (the functional code has a different config than tests)

The tests validate that the ClientConfig that is created in the functional code is equal to the ClientConfig that we generate manually in the tests.

> tests that will just need updated every time default values change
If we add a field to the struct and its default is the zero value, then no. If we change a default, then yes we would need to update the value in the manually created struct in tests.

Reply:
I've updated the tests 👍